src/main/java/com/puppycrawl/tools/checkstyle/utils/BlockCommentPosition.java

```diff
@@ -169,15 +169,21 @@ public static boolean isOnConstructor(DetailAST blockComment) {
      * @return true if node is before enum constant
      */
     public static boolean isOnEnumConstant(DetailAST blockComment) {
-        final boolean isOnPlainConst = blockComment.getParent() != null
-                && blockComment.getParent().getType() == TokenTypes.ENUM_CONSTANT_DEF
-                && getPrevSiblingSkipComments(blockComment).getType() == TokenTypes.ANNOTATIONS
-                && getPrevSiblingSkipComments(blockComment).getChildCount() == 0;
-        final boolean isOnConstWithAnnotation = !isOnPlainConst && blockComment.getParent() != null
-                && blockComment.getParent().getType() == TokenTypes.ANNOTATION
-                && blockComment.getParent().getParent().getParent().getType()
-                    == TokenTypes.ENUM_CONSTANT_DEF;
-        return isOnPlainConst || isOnConstWithAnnotation;
+        final DetailAST parent = blockComment.getParent();
+        boolean result = false;
```

Comment:
unfortunately I am not confident in this refactoring.

Please provide regression report for affected Checks that use this function.

Reply:
Regression for InvalidJavadocPositionCheck, DesignForExtensionCheck, JavadocContentLocationCheck, MissingJavadocPackageCheck 

https://pbludov.github.io/issue-7487-has-children-1/

Reply:
Regression for 

AtclauseOrder, JavadocBlockTagLocation, JavadocParagraph, JavadocTagContinuationIndentation, MissingDeprecated, NonEmptyAtclauseDescription, SingleLineJavadoc, SummaryJavadoc (no real difference):

https://pbludov.github.io/issue-7487-has-children-2/

Reply:
Regression for SummaryJavadoc only (no difference):

https://pbludov.github.io/issue-7487-has-children-3/

Reply:
Please pull latest contribution repo, reports are not mobile friendly, latest report generator do better reports.

Reply:
> https://pbludov.github.io/issue-7487-has-children-2/

is full of differences, please use filter to suppress parse violations.

Reply:
> Please pull latest contribution repo, reports are not mobile friendly, latest report generator do better reports.
```
$ git pull
Already up to date.
```

Reply:
after generation of report, try to make browser window small ... it should render without horizontal scrolling.

Reply:
@pbludov You may have to regenerate difference jar if it doesn't do it itself as it is what creates the report.

Reply:
Regression for SummaryJavadoc only (no difference):

https://pbludov.github.io/issue-7487-has-children-4/

Reply:
Regression for Javadoc checks (no difference):

https://pbludov.github.io/issue-7487-has-children-5/

Reply:
Regression for non-Javadoc checks (no difference):

https://pbludov.github.io/issue-7487-has-children-6/

Reply:
@romani all regression reports are rebuilt.